metapool/notebook_support.py

```diff
@@ -0,0 +1,167 @@
+        curr_study[QIITA_ID_KEY] = curr_qiita_id
+    studies_df = pd.DataFrame(local_studies_info)
+
+    # check for qiita_study_ids in studies_df that aren't in the metadata
+    _check_for_missing_df_ids(studies_df, local_metadata_df, QIITA_ID_KEY,
+                              'studies', 'metadata')
+    # merge the metadata df with the studies_df on the qiita_study_id
+    metadata_plus_df = pd.merge(local_metadata_df, studies_df, on=QIITA_ID_KEY)
+    # pull the qiita study id off the sample name
+    metadata_plus_df[SAMPLE_NAME_KEY] = metadata_plus_df.apply(
+        lambda x: x[SAMPLE_NAME_KEY].replace(f"{x[QIITA_ID_KEY]}.", ""),
+        axis=1
+    )
+
+    # now add the project name and abbreviation to the sample_accession_df
+    # by merging on the qiita_id.sample_name
+    extension_cols = \
+        [SAMPLE_NAME_KEY, PM_PROJECT_NAME_KEY, PM_PROJECT_ABBREV_KEY]
+    _check_for_missing_df_ids(
+        sample_accession_df, metadata_plus_df, SAMPLE_NAME_KEY,
+        'sample accession', 'metadata')
+    sample_accession_plus_df = pd.merge(
+        sample_accession_df, metadata_plus_df[extension_cols],
+        on=SAMPLE_NAME_KEY)
+    return sample_accession_plus_df
+
+
+# add the project abbreviation to the compression layout so the user doesn't
+# have to enter it in two places
+def extend_compression_layout_info(compression_layout, studies_info):
+    extended_compression_layout = compression_layout.copy()
+    # for each dict in compression_layout
+    for curr_plate in extended_compression_layout:
+        # get its Project Name
+        curr_project_name = curr_plate[PM_PROJECT_NAME_KEY]
+        # find that Project Name in studies_info
+        found_study = None
+        for curr_study in studies_info:
+            if curr_study[PM_PROJECT_NAME_KEY] == curr_project_name:
+                found_study = curr_study
+                break
+        # next study
+
+        # add the Project Abbreviation from the studies_info to the
+        # compression layout dict
+        if found_study is None:
+            raise ValueError(f"{PM_PROJECT_NAME_KEY} '{curr_project_name}' "
+                             f"in the compression layout is not in the "
+                             f"studies info")
+        curr_plate[PM_PROJECT_ABBREV_KEY] = found_study[PM_PROJECT_ABBREV_KEY]
+    # next plate
+
+    return extended_compression_layout
+
+
+def _check_for_missing_df_ids(subset_df, superset_df, id_col,
+                              sub_name, sup_name):
+    # if there are any qiita_study_ids in the studies_df that aren't in the
+    # metadata_df, raise an error
+    found_mask = subset_df[id_col].isin(superset_df[id_col])
+    if not found_mask.all():
+        missing_vals = subset_df.loc[~found_mask, id_col].unique()
+        raise ValueError(f'Some {id_col} values in the {sub_name} dataframe '
+                         f'are not in the {sup_name} dataframe: '
+                         f'{", ".join(missing_vals)}')
```

Comment:
Argh, nomenclature rears its ugly head again :D  Here's the thing: although the *repo* is named `metagenomics_pooling_notebook`, that in fact seems to be a misnomer.  The `metapool` module is used all over, in parts of the code that have (AFAICT) no notebook interaction, like `qp_klp.Step`, `qp_klp.klp`, `mg-scripts.Pipeline`, `mg-scripts.ConvertJob`, etc.  I realize that `notebook_support` is a filthy cop-out (as is `util` :D ) ... I kind of like `df_util` because it sounds more specific, but since only 2 of the 3 public functions in this module actually act on dfs (`extend_compression_layout_info` acts only on dictionaries), it is somewhat misleading specificity.  I'm willing to change this to `util`, or discuss other names if you prefer ...

Reply:
😄 Sorry, I know over time it grows increasingly misnamed. 😄 But not adding to that is good, right? I can go with util.py if you're okay with it.

Reply:
done

Reply:
ty!